shopfloor/services/zone_picking.py

```diff
@@ -205,6 +206,12 @@ def _response_for_select_line(
             move_lines, product=product, sublocation=sublocation
         )
         data["confirmation_required"] = confirmation_required
+        placeholder = _("Scan location / pack / product / lot")
+        if self.work.menu.scan_location_or_pack_first and not sublocation:
+            placeholder = _("Scan location / pack")
+        elif sublocation:
+            placeholder = _("Scan product / lot")
+        data["searchbar_placeholder"] = placeholder
```

Comment:
I would expose the flag instead and change the message in the UI only. This is not an exception.

Reply:
hmm this flag shoud be used to check what you scan otherwise what's the point? :thinking:

Reply:
> I would expose the flag instead and change the message in the UI only. This is not an exception.

To me it seems easier to pass the placeholder in the response, instead of generating a flag on the backend, that the frontend will need to decode to generate the placeholder.

Reply:
All the translations for the UI stay in the UI. This shouldn't be an exception.
This very same check you do here, can be done in the UI w/o complicating anything.
Moreover, if you want to display things differently or display another msg you have full control on the UI.

Reply:
Well, it is a fine line between back and front. Like all error/warning messages are translated by the backend but displayed on the frontend.
Anyway I changed this PR see the fixup.